### PR TITLE
use ES6 syntax

### PR DESCRIPTION
**Pull request by davidchambers**
sanctuary-js/sanctuary#687

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -81,90 +81,50 @@
...
 
-} (function(type) {
+}) (type => {
 
   'use strict';
 
   /* istanbul ignore if */
   if (typeof __doctest !== 'undefined') {
-    /* eslint-disable no-unused-vars */
+    /* eslint-disable no-unused-vars, no-var */
```
Comment (Avaq) [inline index.js:90]: Interesting. I never thought to use `var` as a feature. :)

File: index.js [hunk 2]
```diff
@@ -212,71 +172,64 @@
...
 
   //  Value :: Location
-  var Value = 'Value';
+  const Value = 'Value';
 
   //  functionName :: Function -> String
-  var functionName = has ('name', Function) ?
-    function functionName(f) { return f.name; } :
+  const functionName = (
+    has ('name', Function) ?
```
Comment (Avaq) [inline index.js:192]: Do we still need to consider the fallback for function names? Wasn't that an es5-support thing?

Comment (davidchambers) [reply index.js:192]: Good point! You are correct: [ECMAScript 2015 Language Specification § 19.2.4.2][1].


[1]: https://262.ecma-international.org/6.0/#sec-function-instances-name

Comment (davidchambers) [reply index.js:192]: #142 :thinking:

Comment (Avaq) [reply index.js]: Yeah, and depending on the outcome of this https://github.com/sanctuary-js/sanctuary-type-classes/pull/151#discussion_r618284535 discussion, we may end up removing the `functionName` function altogether.

**Review by Avaq — APPROVED** [review]

**Pull request merged**